pkg/basecluster/prepare.go

```diff
@@ -68,19 +74,31 @@ func Prepare(fileName string, validateOnly bool) (clusterName string, modifiedYa
 
 // -----------------------------------------------------------------------------
 
-func removeNamespaceThenEncode(obj runtime.Object, enc *yaml.Encoder) (modified bool, err error) {
+func removeNamespaceThenEncode(obj runtime.Object, enc *yaml.Encoder, casMax, casMin int) (modified bool, err error) {
 	log := logpkg.GetLogger()
 	unstr := &unstructured.Unstructured{}
 	if err := scheme.Scheme.Convert(obj, unstr, nil); err != nil {
 		return false, fmt.Errorf("failed to convert object: %s", err)
 	}
+	kind := unstr.GetKind()
```

Comment:
Move to before line 91, where it's needed. Better yet: if this variable is only used once, let's not declare it, and use `if ustr.GetKind() == "MachineDeployment"`. If it gets used more than once in the future, we can reintroduce the variable.

Reply:
done